NGitLab.Mock/Clients/MilestoneClient.cs

```diff
@@ -14,43 +16,119 @@ public MilestoneClient(ClientContext context, int projectId)
             _projectId = projectId;
         }
 
-        public Models.Milestone this[int id] => throw new System.NotImplementedException();
+        public Models.Milestone this[int id]
+        {
+            get
+            {
+                var project = FindProject(_projectId) ?? throw new InvalidOperationException($"Cannot find project with ID {_projectId}");
```

Comment:
Most of the comments from https://github.com/ubisoft/NGitLab/pull/41 applies here too :)